code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -983,3 +983,39 @@
 
 		visible_message("<span class='notice'>\The [src] attaches \the [O] to \his body!</span>",
 				"<span class='notice'>You attach \the [O] to your body!</span>")
+
+/mob/living/carbon/human/proc/self_diagnostics()
+	set name = "Self-Diagnostics"
+	set desc = "Run an internal self-diagnostic to check for damage."
+	set category = "IC"
+
+	if(stat == DEAD) return
+
+	to_chat(src, "<span class='notice'>Performing self-diagnostic, please wait...</span>")
+	if (do_after(src, 40))
```

Comment:
Just a suggestion because it takes 40 deciseconds(4 seconds), not required. If you add this, it will show progress bar above the src with how much time left to wait. 4 seconds is a lot btw.
```suggestion
	if (do_after(src, 40, act_target = src))
```

Reply:
Nevermind, act_target is src by default